### ISSUE TITLE
I want to use JsonIgnore only when deserializing with System.Text.Json

### ISSUE THREAD

**Issue by neelabo** [dotnet/runtime#34404]
I use Json for the configuration file of the application.
The format of the configuration file changes with the version upgrade of the application, and certain properties may become invalid.
In such cases, we need to make it readable for compatibility with the old configuration file, but we do not want to output to the new configuration file.
I want to be able to do this.

- If it is a `JsonIgnore` attribute, both serialization and deserialization will be invalid.
- With the `IgnoreNullValues` ​​option, properties that you want to output as null will also disappear.
- ["Conditionally ignore a property"](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to?view=netcore-3.1#conditionally-ignore-a-property) is too heavy for the purpose.

Ideally, doing something like `[JsonIgnore(IgnoreDeserialize=false)]` would allow for control of serialization and deserialization.

Please consider it.

**Comment by layomia** [comment]
The serializer's features are designed so that data can roundtrip, i.e serialize -> deserialize -> serialize gives the original result. This is why `JsonIgnoreAttribute` is applied to both serialization and deserialization.

Having a way to specify whether the attribute is applied to serialization/deserialization was considered (see https://github.com/dotnet/runtime/issues/30687#issuecomment-582560531), but it wasn't needed by most people.

A future work-around here is to implement a custom converter which does the desired null-handling specific to reading and writing. This is blocked by https://github.com/dotnet/runtime/issues/34439 (planned for 5.0),
